packages/office-ui-fabric-react/src/utilities/selection/Selection.ts

```diff
@@ -135,37 +135,43 @@ export class Selection implements ISelection {
       }
     }
 
-    if (shouldClear) {
+    if (shouldClear || items.length === 0) {
       this.setAllSelected(false);
     }
 
     // Check the exemption list for discrepencies.
     const newExemptedIndicies: { [key: string]: boolean } = {};
+    let newExemptedCount = 0;
 
     for (const indexProperty in this._exemptedIndices) {
-      if (this._exemptedIndices.hasOwnProperty(indexProperty)) {
+      if (Object.prototype.hasOwnProperty.call(this._exemptedIndices, indexProperty)) {
```

Comment:
uh... this is pretty abnormal. Can you explain? (and if it makes sense to make this change, are there other places to do this? and we probably should comment on abnormal things.)

Reply:
Technically, you can override `hasOwnProperty` within an `Object` sub-type. To to ensure you're doing the correct check, you should really invoke the *real* `hasOwnproperty` function, which is available on `Object.prototype`:
https://stackoverflow.com/questions/12017693/why-use-object-prototype-hasownproperty-callmyobj-prop-instead-of-myobj-hasow

Reply:
Hmm. We have this all over the place. is this a real bug you're fixing? because either:

1. This could hide a bug (nobody should be overriding hasOwnProperty)
2. There are cases when writing tests where mocking this method might be valid, but then wouldn't we want to use the mock?
3. If we accept that people can and should be overriding native methods, which sounds nuts, then shouldn't this be fixed in every place with a tslint rule enforcing the pattern? And even maybe extracted in a utility to centralize, so you don't need to remember the full syntax?

My recommendation would be to not introduce this, or fix the other things listed in 3 above. Doing a one-off here seems to set a really confusing precedent.